### PR TITLE
Frontend Bulk Push

### PR DESCRIPTION
**Pull request by xujunjiejack**
Basically, this front-end version has a lot of new stuff inside.
Couple highlights:
     - We use Babel as new compiler 
     - A new drag and drop design for schedules
     - SSO implemented
     - Better workflow
Other needs to be done before Iteration 2 due:
      - Integrate some tests

### CODE REVIEW

**Comment by cocolocomoco21** [comment]
This cutoff of the navigation (?) bar happens when running: 
image

**Comment by cocolocomoco21** [comment]
"Course Preference" and "Shift Preference" on the tutor view only shows one of these fields. They should show the whole list. Is this easily fixable?

File: electron-src/process/js/viewtutors.js [hunk 1]
```diff
@@ -83,11 +111,32 @@ class MainInterface extends React.Component {
...
+                        {/*<Cour  seTable courses = {this.state.courses}/>*/}
                         {this.prepareView()}
                     </div>
 
-                        <button type="button" className="btn btn-lg btn-success" id="generate-button-pos" onClick={this.ShowViewSchedules} > Generate Schedules! </button>
+                    <button type="button" className="btn btn-lg btn-success" id="generate-button-pos" onClick={this.toggleProceeding} > Generate Schedules! </button>
+
+                    <Modal isOpen={this.state.proceeding}>
+                        <ModalHeader>
+                            Action
```
Comment (cocolocomoco21) [inline electron-src/process/js/viewtutors.js]: Should this be "Generate Schedules" or something, rather than "Action"?

**Comment by cocolocomoco21** [comment]
If possible, the schedule that is shown immediately after generation should be grayed out on the left side multi-selector for schedules, like it is after you select a schedule

**Comment by cocolocomoco21** [comment]
Drag and drop does not work if you are in the shift but not on top of existing tutor-course assignments. For instance, if a shift is empty, it is difficult to add a tutor-course assignment. Also, if you are adding to the bottom of the list of assignments for the shift, dropping below the last assignment does not work.

File: electron-src/process/js/scheduletable.js [hunk 2]
```diff
@@ -35,57 +16,92 @@ class ScheduleTable extends React.Component {
...
-                        <tbody>
-                            {b}
-                        </tbody>
-                    </table>
-                    </div> {/*tool bar */}
+            <div className="container">
+                <div className="row">
+                    <div className="col">
+                        <div className="row" style={{background: "#5bc0de", color: "white", padding: 15}}>
+                            <h3>Schedule: Schedule 1</h3>
```
Comment (cocolocomoco21) [inline electron-src/process/js/scheduletable.js]: This should be dynamic to allow for changing titles depending on which schedule is selected. Not a huge deal for now, but we want this to be dynamic. Not sure if this is the appropriate place to comment this, but I'll leave it here anyway.

**Comment by cocolocomoco21** [comment]
1) Babel (?) or whatever runs the frontend Electron job does not exit, and simply sits at this: 
image

2) The backend Java server process is not killed by Electron/the frontend, and this prevents the user from closing the application and reopening it at any point, unless they restart their computer, since the Java process still runs.
image

**Comment by xujunjiejack** [comment]
What i will fix before due: 
     Fix the schedules toolbar so that it selects something
     Navigation bar
     Java process end.
Potentially fix:
     The multiple courses display.

**Comment by cocolocomoco21** [comment]
Things left to touch up:
- Tutor view - In "Course Preferences," add a space between courses
- Course view - "Preferred Shift Amount" and "Willing Shift Amount", not "Require Shift Amount"
- Generate - After hitting "Generate Schedules" button, pop-up modal should say "Generate Schedules" instead of "Action," and "Do you want to proceed with generating schedules?" instead of "Do you want to proceed the generation?"
- Exit - "Are you sure you want to exit ULCRS?" instead of "Are you sure about exiting the app?"

Otherwise, I think this is good.

**Pull request merged**